node/src/transition_frontier/sync/ledger/snarked/transition_frontier_sync_ledger_snarked_state.rs

```diff
@@ -21,12 +21,19 @@ pub enum TransitionFrontierSyncLedgerSnarkedState {
     Pending {
         time: Timestamp,
         target: SyncLedgerTarget,
-
+        /// Number of accounts in this ledger (as claimed by the Num_accounts query result)
+        num_accounts: u64,
+        /// Number of accounts received and accepted so far
+        num_accounts_accepted: u64,
+        /// Number of hashes received and accepted so far
+        num_hashes_accepted: u64,
+        /// Queue of addresses to query and the expected contents hash
+        queue: VecDeque<LedgerQueryQueued>,
+        /// Pending ongoing address queries and their attempts
         #[serde_as(as = "Vec<(_, _)>")]
-        pending: BTreeMap<LedgerAddress, LedgerQueryPending>,
-        /// `None` means we are done.
-        next_addr: Option<LedgerAddress>,
-        end_addr: LedgerAddress,
+        pending_addresses: BTreeMap<LedgerAddress, LedgerAddressQueryPending>,
+        /// Pending num account query attempts
+        pending_num_accounts: Option<LedgerNumAccountsQueryPending>,
```

Comment:
`num_accounts_expected` + `num_accounts_added` ? The validation of the number of accounts claimed by the peer is kind of weak because at most we validate that the depth of the subtree is correct, but any value for `num_accounts` that matches that depth will be accepted (so it should just be considered an estimate)

Reply:
> Of course assuming that fetching of addresses and num accounts doesn't happen in parallel. If it's sequential, that should be clear based on state.

It is fully sequential (only after we know the number of accounts we know the root node containing all the accounts), so yes, that split makes total sense, will do that

Reply:
I split the Pending state in two (one for the num accounts and another for the merkle tree sync). In your comment above you added some `Success` states, which I didn't add. Can you elaborate on that? should I add them too?

Reply:
Actually, nevermind, it is just cleaner that way, I will add them.